cumulusci/core/flowrunner.py

```diff
@@ -289,6 +289,34 @@ def _rule(self, fill="=", length=60, new_line=False):
         if new_line:
             self.logger.info("")
 
+    def show_summary(self):
+        """ Display the description and steps contained within a given FlowCoordinator """
+        self.logger.info(
+            "Description: {}".format(self.flow_config.config["description"])
+        )
+        previous_step = []
+        step_paths = [step.path.split(".") for step in self.steps]
+        step_num_by_level = [1 for i in range(max(len(path) for path in step_paths))]
+        for step in step_paths:
+            indent = 0
+            for idx, item in enumerate(step):
+                item_type = "task" if idx == len(step) - 1 else "flow"
+                if item not in previous_step:
+                    self.logger.info(
+                        "{}{}) {}: {}".format(
+                            "    " * indent, step_num_by_level[indent], item_type, item
+                        )
+                    )
+                    step_num_by_level[indent] += 1
+                    # when a step num increases we need to
+                    # reset step levels below it to 1
+                    idx2 = indent + 1
+                    while idx2 < len(step_num_by_level):
+                        step_num_by_level[idx2] = 1
+                        idx2 += 1
+                indent += 1
+            previous_step = step
```

Comment:
@Br4nd0R nice work. for fun I wrote my own version which uses the numbers from step.step_num instead of re-counting in step_num_by_level:

```python
        lines = []
        previous_parts = []
        for step in self.steps:
            parts = step.path.split('.')
            numbers = str(step.step_num).split('.')
            task_name = parts.pop()
            for i, flow_name in enumerate(parts):
                if len(previous_parts) < i + 1 or previous_parts[i] != flow_name:
                    lines.append("{}{}) flow: {}".format("    " * i, numbers[i], flow_name))
            lines.append("{}{}) task: {}".format("    " * (i + 1), numbers[-1], task_name))
            previous_parts = parts
        return "\n".join(lines)
```

Your approach looks valid to me though, so please don't feel pressure to adopt anything from my version.

Reply:
I do like how this version iterates directly over self.steps, and how it differentiates between tasks and flows. One thing that I'm running into with this algorithm is that if there is a step_path that just contains a single task at the top level (i.e. `len(parts) == 1`) then the `for` loop that utilizes `enumerate()` never executes and we get an `UnboundLocalError` when we attempt to reference `i` when appending the task name.

Also, we talked briefly about how `step.step_num` isn't consistent throughout all steps (some start with 1 and others with 0) so we will want to work through how to address that too.

Reply:
Good point about step paths that only contain a top-level task. We could avoid that error by initializing `i = -1` before the loop.

Step numbering is intentionally flexible to make it possible to override a flow and insert steps in the middle. So it's possible to use 0, use non-integers, and leave gaps in the sequence. We should preserve those numbers when we display the flow so that it's easy to refer back to the configuration. To deal with this I think we'll want to modify the StepSpec object to store a second number sequence for each step, stored as a list so that we can distinguish between levels. Then we can use this new set of numbers for display purposes.

Another thing I thought of: flow steps can have conditions attached using the "when" clause -- see the `deploy_unmanaged` flow for an example. We should probably display these to make it clear that there is conditional logic involved to determine what actually runs.